rust/src/shards/gui/reset.rs

```diff
@@ -0,0 +1,91 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright © 2022 Fragcolor Pte. Ltd. */
+
+use super::Reset;
+use crate::shard::Shard;
+use crate::shards::gui::util;
+use crate::shards::gui::EGUI_UI_SEQ_TYPE;
+use crate::shards::gui::PARENTS_UI_NAME;
+use crate::types::Context;
+use crate::types::ExposedInfo;
+use crate::types::ExposedTypes;
+use crate::types::ParamVar;
+use crate::types::Types;
+use crate::types::Var;
+use crate::types::ANY_TYPES;
+
+impl Default for Reset {
+  fn default() -> Self {
+    let mut parents = ParamVar::default();
+    parents.set_name(PARENTS_UI_NAME);
+    Self {
+      parents,
+      requiring: Vec::new(),
+    }
+  }
+}
+
+impl Shard for Reset {
+  fn registerName() -> &'static str
+  where
+    Self: Sized,
+  {
+    cstr!("UI.Reset")
+  }
+
+  fn hash() -> u32
+  where
+    Self: Sized,
+  {
+    compile_time_crc32::crc32!("UI.Reset-rust-0x20200101")
+  }
+
+  fn name(&mut self) -> &str {
+    "UI.Reset"
+  }
+
+  fn inputTypes(&mut self) -> &Types {
+    &ANY_TYPES
+  }
+
+  fn outputTypes(&mut self) -> &Types {
+    &ANY_TYPES
+  }
+
+  fn requiredVariables(&mut self) -> Option<&ExposedTypes> {
+    self.requiring.clear();
+
+    // Add UI.Parents to the list of required variables
+    let exp_info = ExposedInfo {
+      exposedType: EGUI_UI_SEQ_TYPE,
+      name: self.parents.get_name(),
+      help: cstr!("The parent UI objects.").into(),
+      ..ExposedInfo::default()
+    };
+    self.requiring.push(exp_info);
+
+    Some(&self.requiring)
+  }
+
+  fn warmup(&mut self, ctx: &Context) -> Result<(), &str> {
+    self.parents.warmup(ctx);
+
+    Ok(())
+  }
+
+  fn cleanup(&mut self) -> Result<(), &str> {
+    self.parents.cleanup();
+
+    Ok(())
+  }
+
+  fn activate(&mut self, _context: &Context, input: &Var) -> Result<Var, &str> {
+    if let Some(ui) = util::get_current_parent(*self.parents.get())? {
+      *ui.ctx().memory() = Default::default();
```

Comment:
What's the utility?
Can't we abstract this when needed?

Reply:
It is in the egui demo and it's immediate effect is to reset all layout change and position (e.g. window position).

This could be useful in the editor to reorganize windows.